test/common/stats/utility_fuzz_test.cc

```diff
@@ -8,8 +12,47 @@ namespace Envoy {
 namespace Fuzz {
 
 DEFINE_FUZZER(const uint8_t* buf, size_t len) {
-  const absl::string_view string_buffer(reinterpret_cast<const char*>(buf), len);
-  Stats::Utility::sanitizeStatsName(string_buffer);
+  {
+    const absl::string_view string_buffer(reinterpret_cast<const char*>(buf), len);
+    Stats::Utility::sanitizeStatsName(string_buffer);
+  }
+
+  // generate a random number as the maximum length of the stat name
+  const size_t max_len = *reinterpret_cast<const uint8_t*>(buf) % len;
```

Comment:
I really appreciate that you pointed out a bug for me here. But I am confused why `*buf` should be mod-ed by `(len - 1)` rather than `len`? When `len` is 1, `(len - 1)` will cause crashes. Or do you mean `(*buf % len) + 1`?

Reply:
If your goal is to get a random 8-bit number between [0,len) then the code looks good other than the early exit. But what you want is a between [1, len) then you can use `1 + (*buf % (len - 1))`.

It looks like you could simplify the logic below by going for a number of bytes from [1,len-1).

actually, looking below, you might want to go for a minimum length of 2 or 3 so you can use more bytes for control without worrying about special-cases.

Reply:
Make sense to me. Thank you for explaining!